fendermint/vm/topdown/src/sync.rs

```diff
@@ -231,53 +241,77 @@ async fn reset_cache<T: ParentFinalityStateQuery + Send + Sync + 'static>(
 /// Obtains the last recorded height from provider cache or from last committed finality height.
 async fn last_recorded_height(
     provider: &Arc<Toggle<CachedFinalityProvider<IPCProviderProxy>>>,
-) -> anyhow::Result<Option<BlockHeight>> {
-    let height = atomically(|| {
-        let h = if let Some(h) = provider.latest_height()? {
+) -> anyhow::Result<Option<(BlockHeight, BlockHash)>> {
+    let result = atomically(|| {
+        let h = if let Some(h) = provider.latest_height_hash()? {
             Some(h)
         } else if let Some(f) = provider.last_committed_finality()? {
-            Some(f.height)
+            ome((f.height, f.block_hash))
         } else {
             None
         };
         Ok(h)
     })
     .await;
 
-    Ok(height)
+    Ok(result)
 }
 
 /// Obtain the new parent views for the input block height range
 async fn get_new_parent_views(
     parent_proxy: &Arc<IPCProviderProxy>,
+    previous_hash: BlockHash,
```

Comment:
If you want you can just declare this as `mut previous_hash: BlockHash,` and then the `let mut previous_hash = previous_hash;` is not needed.